sdk/python/kfp/deprecated/_client.py

```diff
@@ -1161,6 +1162,167 @@ def enable_job(self, job_id: str):
+                    experiment_id = resource_reference.key.id
+                    break
+
+        if not job_name:
+            job_name = f"Clone of {job.name}"
+
+        if not description:
+            description = job.description
+
+        if not start_time:
+            if job.trigger.cron_schedule:
+                start_time = job.trigger.cron_schedule.start_time
+            else:
+                start_time = job.trigger.periodic_schedule.start_time
+
+        if not end_time:
+            if job.trigger.cron_schedule:
+                end_time = job.trigger.cron_schedule.end_time
+            else:
+                end_time = job.trigger.periodic_schedule.end_time
+
+        if not interval_second:
+            if job.trigger.periodic_schedule:
+                interval_second = job.trigger.periodic_schedule.interval_second
+
+        if not cron_expression:
+            if job.trigger.cron_schedule:
+                cron_expression = job.trigger.cron_schedule.cron
+
+        if not max_concurrency:
+            max_concurrency = job.max_concurrency
+
+        if not no_catchup:
+            no_catchup = job.no_catchup
+
+        if not params:
+            params = job.pipeline_spec.parameters[0]
+
+        if not pipeline_package_path:
+            manifest = json.loads(job.pipeline_spec.workflow_manifest)
+            pipeline_package_path="/tmp/clone_recurring_run.yaml"
+            with open(pipeline_package_path, "w") as f:
+                yaml.dump(manifest, f, allow_unicode=True)
+
+        if not pipeline_id:
+            pipeline_id = job.pipeline_spec.pipeline_id
+
+        if not version_id:
+            for resource_reference in job.resource_references:
+                if resource_reference.key.type == "PIPELINE_VERSION":
+                    version_id = resource_reference.id
+                    break 
+
+        if not enabled:
+            enabled = job.enabled
+
+        # Note, we don't override enable_caching since it's not configurable
+        # from the GUI and is applied to individual templates rather than the
+        # workflow as a whole. 
+
+        if not service_account:
+            service_account = job.service_account
+
+        cloned_recurring_run = self.create_recurring_run(
+            experiment_id=experiment_id,
+            job_name=job_name,
+            description=description,
+            start_time=start_time,
+            end_time=end_time,
+            interval_second=interval_second,
+            cron_expression=cron_expression,
+            max_concurrency=max_concurrency,
+            no_catchup=no_catchup,
+            params=params,
+            pipeline_package_path=pipeline_package_path,
+            pipeline_id=pipeline_id,
+            version_id=version_id,
+            enabled=enabled,
+            enable_caching=enable_caching,
+            service_account=service_account,
+        )
+
+        # Cleanup workflow manifest.
+        os.remove(pipeline_package_path)
```

Comment:
Consider using a context manager for this, in case an exception is thrown between file creation and cleanup.

Reply:
Thanks for the spot on feedback, @connor-mccarthy! All comments addressed. Marking conversations as resolved.